expression/core/option.py

```diff
@@ -329,8 +329,8 @@ def validate_none(v: Any, handler: ValidatorFunctionWrapHandler) -> Option[Any]:
                 ),
                 core_schema.chain_schema(
                     [
-                        # Ensure the value is an instance of _T
-                        core_schema.is_instance_schema(item_tp),
+                        # item_tp's schema should ensure the value is an instance of _T
+                        # is_instance_schema doesn't work for Annotated[_T, ...]
```

Comment:
I need to better understand what we lose by removing this line. What non-valid code would now be possible to write that was not possible before?

Reply:
This seems to be a simple issue that will be resolved once this issue is fixed.
https://github.com/python/cpython/issues/113702

Until then, using the method written in the issue seems like a good idea.

```python
# https://github.com/python/cpython/issues/113702
# Workaround for python 3.11 and 3.12
from typing import Any, Annotated, Required, NotRequired, get_origin, get_args

def fetch_annotations(type_var) -> tuple[Any, ...]:
    if get_origin(type_var) in {Annotated, Required, NotRequired}:  # 👈 This set is a problem
        child_annotations = fetch_annotations(get_args(type_var)[0])
        if get_origin(type_var) is Annotated:
            return get_args(type_var)[1:] + child_annotations
        return child_annotations
    return ()
```

It can be resolved by replacing the following areas
https://github.com/dbrattli/Expression/blob/d4bd6c762e849f7709c4752c1722d8b13b58accd/expression/core/option.py#L303-L308

Reply:
By removing this line, there could a case that if the internal type `T`'s validator doesn't return the type `T `after validating. Then we will not get a `Option[T]` but instead what the validator of `T` returns.
And yes @phi-friday is a possible solution,  but I think it is somehow unnecessary though it could be safer.

Reply:
I created the pr that is compatible with `python==3.10`, just in case we need it. 
Please review #222  if needed.

Reply:
It doesn't seem to be enough to comment it out, as it fails to create the class if it is nested like this
```python
Result[Annotated[Annotated[CustomString, 1], 2], Any]
```

Reply:
I did some more checking, and it looks like I was wrong.

The `handler` of `validate_some` checks for things related to `item_tp` as well,
so I think it's okay to comment it out as @denghz  said.

I apologize for the confusion.